tests/components/konnected/test_config_flow.py

```diff
@@ -0,0 +1,1052 @@
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "confirm"
+    assert result["description_placeholders"] == {
+        "model": "Konnected Alarm Panel",
+        "host": "1.2.3.4",
+        "port": 1234,
+    }
+
+
+async def test_import_no_host_user_finish(hass, mock_panel):
+    """Test importing a panel with no host info."""
+    mock_panel.get_status.return_value = {
+        "mac": "11:22:33:44:55:66",
+        "name": "Konnected Pro",
+    }
+
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN,
+        context={"source": "import"},
+        data={
+            "default_options": {
+                "blink": True,
+                "discovery": True,
+                "io": {
+                    "1": "Disabled",
+                    "10": "Disabled",
+                    "11": "Disabled",
+                    "12": "Disabled",
+                    "2": "Disabled",
+                    "3": "Disabled",
+                    "4": "Disabled",
+                    "5": "Disabled",
+                    "6": "Disabled",
+                    "7": "Disabled",
+                    "8": "Disabled",
+                    "9": "Disabled",
+                    "alarm1": "Disabled",
+                    "alarm2_out2": "Disabled",
+                    "out": "Disabled",
+                    "out1": "Disabled",
+                },
+            },
+            "id": "aabbccddeeff",
+        },
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+
+    # confirm user is prompted to enter host
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={"host": "1.1.1.1", "port": 1234}
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "confirm"
+    assert result["description_placeholders"] == {
+        "model": "Konnected Alarm Panel Pro",
+        "host": "1.1.1.1",
+        "port": 1234,
+    }
+
+    # final confirmation
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={}
+    )
+    assert result["type"] == "create_entry"
+
+
+async def test_ssdp_already_configured(hass, mock_panel):
+    """Test if a discovered panel has already been configured."""
+    MockConfigEntry(
+        domain="konnected",
+        data={"host": "0.0.0.0", "port": 1234},
+        unique_id="112233445566",
+    ).add_to_hass(hass)
+    mock_panel.get_status.return_value = {
+        "mac": "11:22:33:44:55:66",
+        "name": "Konnected Pro",
+    }
+
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN,
+        context={"source": "ssdp"},
+        data={
+            "ssdp_location": "http://0.0.0.0:1234/Device.xml",
+            "manufacturer": config_flow.KONN_MANUFACTURER,
+            "modelName": config_flow.KONN_MODEL_PRO,
+        },
+    )
+    assert result["type"] == "abort"
+    assert result["reason"] == "already_configured"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4986898)